File: c/src/basic/union.c
#include "union.h"


void union_test(){
  union Dato v;
  v.intero = 10;
  printf( "Intero: %d\n"  , v.intero ) ;
  v.decimale = 3.14;
  printf( "Decimale: %f\n", v.decimale ) ;
  v.carattere = 'A';
  printf( "Carattere: %c\n", v.carattere ) ;
}

File: rust/src/basic/union.rs

union Valore {
  intero: i32,
  decimale: f32,
  carattere: char,
}

pub fn union_test() {
  let mut v = Valore { intero: 10 }; //  del campo `intero`.
  unsafe { // scrivere i campi della `union` richiede un blocco `unsafe`
    println!("Intero: {}", v.intero);
    v.decimale = 3.14;
    println!("Decimale: {}", v.decimale);
    v.carattere = 'A';
    println!("Carattere: {}", v.carattere);
  }
}
